### PR TITLE
EVG-7425: allow non-running hosts to assign running task

### PR DESCRIPTION
**Pull request by Kimchelly**

### CODE REVIEW

File: model/host/host.go [hunk 1]
```diff
@@ -989,7 +989,7 @@ func (h *Host) UpdateRunningTask(t *task.Task) (bool, error) {
 
 	selector := bson.M{
 		IdKey:          h.Id,
-		StatusKey:      evergreen.HostRunning,
+		StatusKey:      bson.M{"$in": []string{evergreen.HostProvisioning, evergreen.HostRunning}},
```
Comment (john-m-liu) [inline model/host/host.go]: Do we also want to allow HostStarting here? I think when I was investigating some hosts were in this state and asking for tasks

Comment (ybrill) [reply model/host/host.go]: 👍 isn't this the main issue (given that hosts don't go into provisioning until DescribeInstances tells us they're running)?

Comment (Kimchelly) [reply model/host/host.go]: Done.

Comment (john-m-liu) [reply model/host/host.go]: Do we want to allow all those statuses though? I can't think of how we might get into the situation, but it seems weird that we'd let an uninitialized host run a task

Comment (Kimchelly) [reply model/host/host.go]: I only did this to be consistent with Sam's original PR. However, I don't really want to allow other statuses so I'm going to ban all statuses except starting, provisioning, and running and only for the user data case.

File: service/api_task.go [hunk 2]
```diff
@@ -110,7 +110,13 @@ func checkHostHealth(h *host.Host) bool {
 		return false
 	}
 
-	if h.Distro.BootstrapSettings.Method == distro.BootstrapMethodUserData && !util.StringSliceContains(evergreen.DownHostStatus, h.Status) {
+	// User data can start anytime after the instance is created, so the app
+	// server may not have marked it as running yet.
+	if h.Distro.BootstrapSettings.Method == distro.BootstrapMethodUserData &&
+		!util.StringSliceContains([]string{
```
Comment (ybrill) [inline service/api_task.go]: I would think we'd want to check if it does contain one of these statuses

File: service/api_task_test.go [hunk 3]
```diff
@@ -775,7 +775,7 @@ func TestNextTask(t *testing.T) {
 				})
 				Convey("should exit when it is quarantined", func() {
 					So(nonLegacyHost.SetStatus(evergreen.HostQuarantined, evergreen.User, ""), ShouldBeNil)
-					So(db.Update(host.Collection, bson.M{host.IdKey: nonLegacyHost.Id}, bson.M{"$set": bson.M{host.StatusKey: evergreen.HostQuarantined}}), ShouldBeNil)
+					nonLegacyHost.Status = evergreen.HostQuarantined
```
Comment (ybrill) [inline service/api_task_test.go]: isn't this accomplished by SetStatus?

**Pull request merged**